src/tools/idlc/src/idlc.c

```diff
@@ -466,6 +469,9 @@ static const idlc_option_t *compopts[] = {
   &(idlc_option_t){
     IDLC_FLAG, { .flag = &config.version }, 'v', "", "",
     "Display version information." },
+  &(idlc_option_t){
+    IDLC_STRING, { .string = &config.export_macro }, 'e', "", "<export macro>",
+    "Add export macro before topic descriptors." },
```

Comment:
Yes, this export macro option is backend specific (only needed for C probably?), This option is needed to generate the correct macros for the built-in types used in the XTypes type system (included in DDSI, e.g. [ddsi_xt_typeinfo.h](https://github.com/eclipse-cyclonedds/cyclonedds/blob/26177c6cf1d8f84b8ccf96cb6af177206677eb09/src/core/ddsi/include/dds/ddsi/ddsi_xt_typeinfo.h#L1280) in PR #904) But for other cases it could also be useful I think, e.g. when using a shared library for generated types. 

For the C++ back-end, we'll also need to generate the XTypes type object cdr blobs in the topic descriptors to get type assignability checking, but that is not related to this idlc option.

Reply:
We may want think about exporting command line options to backends too. e.g. #922 adds the `-d` option that we want to share between backends but currently can't because the options are not exported.